### PR TITLE
adding ec2 tag and metadata as part of labels

### PR DESCRIPTION
**Pull request by cch0**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,10 +9,32 @@ metrics in JSON format via HTTP POST as sent by collectd's
...
-This exporter is useful for exporting metrics from existing collectd setups, as
+This is useful for exporting metrics from existing collectd setups, as
 well as for metrics which are not covered by the core Prometheus exporters such
 as the [Node Exporter](https://github.com/prometheus/node_exporter).
 
+## AWS EC2 Specifics
+
+This exporter is specially built for reporting AWS EC2 instance metrics.
+
+The exporter will periodically calls EC2 Introspection API as well as EC2 Describe Tags API to pull the latest EC2 metadata and tag information.
```
Comment (fastnsilver) [inline README.md]: `calls` -> `call`. or remove `will` and add `the` before `EC2 ...`

File: README.md [hunk 2]
```diff
@@ -12,10 +9,32 @@ metrics in JSON format via HTTP POST as sent by collectd's
...
+
+These metadata and tag information will then first be santized according to the Prometheus metric names and labels [naming convention](https://prometheus.io/docs/concepts/data_model/) and made part of ingested data as values for labels.
```
Comment (fastnsilver) [inline README.md]: sanitized

File: README.md [hunk 3]
```diff
@@ -55,22 +74,53 @@ To change the path of the end-point, use the `-web.collectd-push-path` command
...
+
+To run the executable binary from the command line
+
+```$./collectd_exporter```
+
+To run the Docker container
+
+```docker run -d  -p 9103:9103 -p 25826:25826/udp tmobile/collectd-exporter -collectd.listen-address=":25826" -metadata.refresh.period.min="1"```
 
-```bash
```
Comment (fastnsilver) [inline README.md:66]: We're going to add this back once we've populated T-mo's Artifactory?

File: main.go [hunk 4]
```diff
@@ -76,33 +81,60 @@ func newName(vl api.ValueList, index int) string {
 }
 
 // newLabels converts the plugin and type instance of vl to a set of prometheus.Labels.
-func newLabels(vl api.ValueList) prometheus.Labels {
+func newLabels(vl api.ValueList, md metadata) prometheus.Labels {
```
Comment (fastnsilver) [inline main.go:84]: You sneaky devil.

File: README.md (hunk 3 above)
Comment (cch0) [reply README.md:66]: Depends on how this binary going to be executed. Running it as binary is possible but not as good as running it as docker container assuming all their ec2 instances can run docker. I will add the docker section if that is the direction we are going with in the future.

File: README.md (hunk 2 above)
Comment (cch0) [reply README.md]: done

File: README.md (hunk 1 above)
Comment (cch0) [reply README.md]: done

**Pull request merged**